packages/parse5/lib/parser/index.test.ts

```diff
@@ -110,4 +111,41 @@ describe('parser', () => {
             expect(onItemPop).toHaveBeenLastCalledWith(bodyElement.childNodes[0], bodyElement);
         });
     });
+
+    describe('rawtext parsing', () => {
+        it.each([
+            ['iframe'],
+            ['noembed'],
+            ['noframes'],
+            ['noscript'],
+            ['script'],
+            ['style'],
+            ['textarea'],
+            ['title'],
+            ['xmp'],
+        ])('<%s>', (tagName) => {
+            const html = `<r><${tagName}><math id="</${tagName}><b>should be outside</b>">`;
+            const fragment = parseFragment(html);
+
+            expect(fragment.childNodes.length).toBe(1);
+            const r = fragment.childNodes[0] as Element;
+            expect(r.nodeName).toBe('r');
+            expect(r.childNodes).toHaveLength(3);
+            expect(r.childNodes.map(_ => _.nodeName)).toEqual([
+                tagName,
+                'b',
+                '#text'
+            ]);
+
+            const target = r.childNodes[0] as Element;
+            expect(target.childNodes).toHaveLength(1);
+            expect(target.childNodes[0].nodeName).toBe('#text');
+            expect((target.childNodes[0] as TextNode).value).toBe('<math id="');
+
+            const b = r.childNodes[1] as Element;
+            expect(b.childNodes).toHaveLength(1);
+            expect(b.childNodes[0].nodeName).toBe('#text');
+            expect((b.childNodes[0] as TextNode).value).toBe('should be outside');
+        });
+    });
```

Comment:
I couldn't find a better place to put this test. For some reason the tests in `test/data/html5lib-tests/serializer` do not repro the issue.